lighthouse-core/gather/drivers/driver.js

```diff
@@ -447,6 +478,30 @@ class Driver {
       storageTypes: 'all',
     });
   }
+
+  /**
+   * Keeps track of calls to a JS function and returns a list of {url, line, col}
+   * of the usage. Should be called before page load (in beforePass).
+   * @param {string} funcName The function name to track ('Date.now', 'console.time').
+   * @param {string} globalVarToPopulate The variable name to populate with stack traces.
+   *     This should unique and on the global object ('window.__dateNowStackTraces').
+   * @return {function(): !Promise<!Array<{url: string, line: number, col: number}>>}
+   *     Call this method when you want results.
+   */
+  captureFunctionCallSites(funcName, globalVarToPopulate) {
```

Comment:
why not just, like, ``const globalVarToPopulate = `window['__${funcName}StackTraces']` `` and then you'll get it for free and you'll be sure no one else will be overwriting it :)